### ISSUE TITLE
[rb] Failed to set preferences: Unable to read profile preferences file

### ISSUE THREAD

**Issue by anhkhang2304** [SeleniumHQ/selenium#8845]
## 🐛 Bug Report

An error occurred while initializing a firefox driver.

## To Reproduce
```
profile = Selenium::WebDriver::Firefox::Profile.new
options = Selenium::WebDriver::Firefox::Options.new(profile: profile)
driver = Selenium::WebDriver.for :firefox, options: options
driver.navigate.to "https://www.google.com"
```
## Error
```
Failed to set preferences: Unable to read profile preferences file (Selenium::WebDriver::Error::SessionNotCreatedError)
```

Detailed steps to reproduce the behavior:

## Expected behavior

I want to run the test on latest versions but I can't seem to pass profile argument to options. Although the document has not changed anything

## Environment

OS: ubuntu 18.04
Browser: Firefox
Browser version: 72.0.1
Browser Driver version: GeckoDriver 0.26.0 
Language Bindings version: ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-linux]
Selenium Webdriver: 4.0.0.alpha6

**Comment by AutomatedTester** [comment]
@raju249 

Please look at the following things in this issue

- [ ] Check that profile is passed in properly

- [ ] We should deprecate `Profile` as people should use the `add_preference` API and `install_addon` APIs on `Options` instead when being passed to geckodriver (APIs are called that in python so might be slightly different)

**Comment by titusfortner** [comment]
You can see some background on what can and can't be done here: https://github.com/SeleniumHQ/selenium/issues/7796

@AutomatedTester  The main reason to use the Profile class is to pass in a zipped profile, which geckodriver is [supposed to support](https://firefox-source-docs.mozilla.org/testing/geckodriver/Profiles.html), but, unfortunately, [does not support](https://github.com/mozilla/geckodriver/issues/1656)

The main use case is that you can not currently set certificates or extensions as part of the capabilities.

**Comment by titusfortner** [comment]
Additional info on state of Ruby's Firefox Profile class: https://github.com/SeleniumHQ/selenium/pull/7376

Theoretically the class should be able to:
1. Zip an existing profile to send to capabilities (which, as I said, [doesn't work](https://github.com/mozilla/geckodriver/issues/1656))
2. Create a default profile that lets you add extensions (since you [can't do that in capabilities](https://github.com/mozilla/geckodriver/issues/1476))

I think this works locally, but not remotely, but I haven't really investigated (which is what this was for: https://github.com/SeleniumHQ/selenium/issues/7404)

**Comment by twalpole** [comment]
This is because the profile being generated is invalid due to homepage_welcome_url being set to homepage - even when not set.  Short term workaround would be to include a setting for `browser.startup.homepage` in your specified profile until a new gem is released

**Comment by twalpole** [comment]
Fixed in PR# 9068

**Comment by titusfortner** [comment]
Merged fix with #9138
